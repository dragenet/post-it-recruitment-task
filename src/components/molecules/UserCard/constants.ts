import { User } from '~/api';

export interface UserCardProps {
  user: User;
}
