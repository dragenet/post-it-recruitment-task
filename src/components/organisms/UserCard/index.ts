export * from './UserCard';
export * from './constants';
