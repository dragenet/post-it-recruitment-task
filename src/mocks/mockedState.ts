import { AppState } from '~/store';
import { mockedUsers } from '~/mocks/mockedUsers';
import { mockedPosts } from '~/mocks/mockedPosts';

export const mockedState: AppState = {
  users: [mockedUsers[0]],
  posts: [mockedPosts[0]],
  currentUser: null,
};
