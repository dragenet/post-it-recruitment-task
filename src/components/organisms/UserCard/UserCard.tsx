import {
  Box,
  Card,
  CardContent,
  Chip,
  Container,
  Grid,
  Link,
  Stack,
  Typography,
} from '@mui/material';
import EmailIcon from '@mui/icons-material/Email';
import PhoneIcon from '@mui/icons-material/Phone';
import LanguageIcon from '@mui/icons-material/Language';

import * as C from './constants';

export const UserCard = ({ user }: C.UserCardProps) => (
  <Card sx={{ height: '100%' }}>
    <CardContent>
      <Box marginBottom={1}>
        <Typography variant="subtitle2" color="text.secondary">
          @{user.username}
        </Typography>
        <Typography variant="h4">{user.name}</Typography>
      </Box>

      <Typography variant="h5" marginBottom={0.5}>
        Contact
      </Typography>
      <Container sx={{ marginBottom: 1 }}>
        <Stack direction="row" alignItems="center" gap={1}>
          <PhoneIcon />
          <Typography>{user.phone}</Typography>
        </Stack>

        <Stack direction="row" alignItems="center" gap={1}>
          <EmailIcon />
          <Link href={`mailto:${user.email}`}>{user.email}</Link>
        </Stack>

        <Stack direction="row" alignItems="center" gap={1}>
          <LanguageIcon />
          <Link href={user.website}>{user.website}</Link>
        </Stack>
      </Container>

      <Typography variant="h5" marginBottom={0.5}>
        Company
      </Typography>
      <Container>
        <Typography variant="h6">{user.company.name}</Typography>
        <Typography color="text.secondary">
          {user.company.catchPhrase}
        </Typography>
        <Grid container direction="row" marginTop={0.5} spacing={0.5}>
          {user.company.bs.split(' ').map((item) => (
            <Grid key={item} item>
              <Chip size="small" label={item} />
            </Grid>
          ))}
        </Grid>
      </Container>
    </CardContent>
  </Card>
);
