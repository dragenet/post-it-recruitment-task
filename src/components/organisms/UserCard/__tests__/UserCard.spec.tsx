import { render } from '@testing-library/react';
import { UserCard } from '../UserCard';
import { mockedUsers } from '~/mocks/mockedUsers';

describe('UserCard', () => {
  it('renders correctly with mocked user', () => {
    const { container } = render(<UserCard user={mockedUsers[0]} />);
    expect(container).toMatchSnapshot();
  });
});
