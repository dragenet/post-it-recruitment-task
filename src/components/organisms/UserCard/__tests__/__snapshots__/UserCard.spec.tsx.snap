// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard renders correctly with mocked user 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-zwt9xa-MuiPaper-root-MuiCard-root"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          class="MuiBox-root css-19idom"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-hav9kw-MuiTypography-root"
          >
            @
            Bret
          </h6>
          <h4
            class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
          >
            Leanne Graham
          </h4>
        </div>
        <h5
          class="MuiTypography-root MuiTypography-h5 css-1su4552-MuiTypography-root"
        >
          Contact
        </h5>
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-s9j4q4-MuiContainer-root"
        >
          <div
            class="css-1rlitzi-MuiStack-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="PhoneIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              1-770-736-8031 x56442
            </p>
          </div>
          <div
            class="css-1rlitzi-MuiStack-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="EmailIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
                href="mailto:Sincere@april.biz"
              >
                Sincere@april.biz
              </a>
            </p>
          </div>
          <div
            class="css-1rlitzi-MuiStack-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LanguageIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
                href="hildegard.org"
              >
                hildegard.org
              </a>
            </p>
          </div>
        </div>
        <h5
          class="MuiTypography-root MuiTypography-h5 css-1su4552-MuiTypography-root"
        >
          Company
        </h5>
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Romaguera-Crona
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
          >
            Multi-layered client-server neural-net
          </p>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-0.5 css-15aqvcf-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-1fj5pxn-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                >
                  harness
                </span>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-1fj5pxn-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                >
                  real-time
                </span>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-1fj5pxn-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                >
                  e-markets
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
